### PR TITLE
[wled] Fix: FX and Palettes channel StateDescriptionProvider handling

### LINKED ISSUE #17276: [wled] FX and Palettes channel StateDescriptionProviders not always updated
If you start openHAB while the WLED controller is OFFLINE, the `Thinghandler` calls `initialize` without being able to populate the FX and Palette channels with all available settings from the controller.

If the `BridgeHandler` notices that the WLED controller is available, it goes ONLINE and so does the `ThingHandler`. However, the `ThingHandler` does not fill these channels and thus they stay empty until one manually deactivates and reactivates the thing.

### PR DESCRIPTION
**Pull request by t2000**
Fixes #17276

### CODE REVIEW

**Comment by t2000** [comment]
PS: It would also be nice to see some progress on https://github.com/openhab/openhab-addons/pull/16907

Its not really motivating (although I understand that it is summer holiday season) to see small bugfixes hanging around for 2 months :-(

**Comment by lsiepel** [comment]
> PS: It would also be nice to see some progress on #16907
> 
> Its not really motivating (although I understand that it is summer holiday season) to see small bugfixes hanging around for 2 months :-(

That PR was merged now. We prefer to wait for codeowners to comment, but unfortunately many are time constrained. Thank you for your contribution. If you have a good understanding of the binding you can always apply as codeowner.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by lolodomo** [comment]
Build failed, please check the log output.

File: bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java [hunk 1]
```diff
@@ -258,4 +256,33 @@ public void initialize() {
...
+                WledApi localAPI = localBridgeHandler.api;
+                if (localAPI != null) {
+                    localBridgeHandler.stateDescriptionProvider.setStateOptions(
+                            new ChannelUID(getThing().getUID(), CHANNEL_FX), localAPI.getUpdatedFxList());
+                    localBridgeHandler.stateDescriptionProvider.setStateOptions(
+                            new ChannelUID(getThing().getUID(), CHANNEL_PALETTES), localAPI.getUpdatedPaletteList());
+                }
+            }
+        }
+
```
Comment (lolodomo) [inline bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java:274]: The build might fail due to this unexpected blank line.

File: bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java [hunk 2]
```diff
@@ -258,4 +256,33 @@ public void initialize() {
...
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        super.bridgeStatusChanged(bridgeStatusInfo);
+
+        if (ThingStatus.ONLINE.equals(bridgeStatusInfo.getStatus())) {
+            // if the handler has been started before the WLED controller is available, we have to fill the providers
+            // again once the bridge goes ONLINE
+            updateStateDescriptionProviders();
+        }
+
```
Comment (lolodomo) [inline bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java:286]: Also here.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java (hunk 1 above)
Comment (t2000) [reply bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java]: Thanks, I have removed it.

File: bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java (hunk 2 above)
Comment (t2000) [reply bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java]: Thanks, I have removed it.

**Review by lolodomo — APPROVED** [review]

**Pull request merged**